### PR TITLE
Changes to the in-race chat (don't crash server)

### PR DESCRIPTION
**Pull request by bls1999**
I threw everything in a try/catch and added a new `foreach`, using `$this->player_array as $player` before calling `$player->socket->write()`. Hopefully that fixes things.

### CODE REVIEW

**Comment by LP-PR2** [comment]
Hey guys let's have a staff guild server to test stuff and not use it.. brilliant planning

**Comment by bls1999** [comment]
@LP-PR2 how are we supposed to test code before it goes into the game?

**Comment by LP-PR2** [comment]
@bls1999 if you're going to request a private server for "testing" purposes and not pulse the updates onto that server and test for full functionality, what is the point of having a staff only private server?

**Comment by bls1999** [comment]
@LP-PR2 I'm not the one pushing the updates. I can't control what gets pushed when.

**Comment by LP-PR2** [comment]
@bls1999 maybe you should have thought about that before requesting your server and maybe hinted at the idea that the new updates are strictly ran on this server to avoid gamebreaking errors like this one and the many other previous ones.

**Comment by fourst4r** [comment]
Pull requests should be independent of each other. I suggest you have a read here: http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/

**Comment by bls1999** [comment]
@LP-PR2 "When we test changes on an open server, it may open the server to crashes. Then we get posts like # 138 which accuse us of intentionally breaking servers, as well as triggering some happy hours on server restart." I think that's pretty clear that we want to test changes on a private server.

@fourst4r This pull has one commit. :thinking:

**Comment by bls1999** [comment]
@fourst4r then why bring it up on this pull request?

**Comment by fourst4r** [comment]
@bls1999 Just please read the blog post...

**Comment by bls1999** [comment]
@fourst4r thank you for the resource. I will think about the content and possibly apply it to my pulls.

**Comment by jacob-grahn** [comment]
I was the one who merged the change into all of the servers, rather than just the testing server

File: multiplayer_server/rooms/Game.php [hunk 1]
```diff
@@ -859,23 +859,32 @@ public function set_var($player, $data)
...
-            if (!$player->is_ignored_id($user_id)) {
-                $player->socket->write("chat`$chat_message");
+        try {
+            $chat_message = substr($message, 5);
+            
+            // check if trying to execute a socket command other than chat
+            if (strpos($message, 'chat`') !== 0) {
+                throw new Exception('Illegal operation.');
+            } // check for backticks
+            if (strpos($chat_message, '`') !== false) {
```
[the remark this answers is not in the document]
Comment (EternalUF) [reply multiplayer_server/rooms/Game.php]: Yup! Tying in race results in “Illegal character in message” now

**Comment by bls1999** [comment]
Oh, great! :/

**Pull request merged**